### ISSUE TITLE
Maybe: Error on Find Masturbation Bucket

### ISSUE THREAD

**Issue by vegapnk** [vegapnk/Cumpilation#3]
Message on Discord: 
`Getting another issue, seems to be a conflict w/ sexperience? I didn't notice what was happening when it happened, sorry. `
[Hugslog](https://discord.com/channels/374305025486225409/1283821424749973545/1284988812602245252)

Someone mentioned this error did happen somewhere already, before Cumpilation. 
But maybe not, so I have to find out...

## Screenshots
![image](https://github.com/user-attachments/assets/30f3a945-6f47-45e0-9dfa-0b9b7999f8ae)
![image](https://github.com/user-attachments/assets/5022037a-c458-4f38-8dc7-3b157146f1cf)

**Comment by vegapnk** [comment]
I got some more errors regarding this: 

```
Exception in rjw.ThinkNode_ChancePerHour_Fappin TryIssueJobPackage: System.NullReferenceException: Object reference not set to an instance of an object
[Ref 928B800C]
 at rjw.ISexPartHediffExtensions.GetPartComp (rjw.ISexPartHediff partHediff) [0x00000] in <36fdf7cfc73c41228e347ef9cdc14d0d>:0 
 at Cumpilation.Gathering.Patch_CasualSexHelper_FindNearbyBucketToMasturbate.Prefix (Verse.Pawn pawn, Verse.Pawn partner, Verse.IntVec3& __result) [0x000c9] in <59f6c4056e6f4708a47ba91679f74dbb>:0 
 at rjw.CasualSex_Helper.FindSexLocation (Verse.Pawn pawn, Verse.Pawn partner) [0x00024] in <36fdf7cfc73c41228e347ef9cdc14d0d>:0 
     - PREFIX vegapnk.cumpilation: Boolean Cumpilation.Gathering.Patch_CasualSexHelper_FindNearbyBucketToMasturbate:Prefix(Pawn pawn, Pawn partner, IntVec3& __result)
 at rjw.JobGiver_Masturbate.TryGiveJob (Verse.Pawn pawn) [0x000f2] in <36fdf7cfc73c41228e347ef9cdc14d0d>:0 
 at Verse.AI.ThinkNode_JobGiver.TryIssueJobPackage (Verse.Pawn pawn, Verse.AI.JobIssueParams jobParams) [0x00000] in <2464551dcee346df9bf15b1c06ef09e9>:0 
 at Verse.AI.ThinkNode_Priority.TryIssueJobPackage (Verse.Pawn pawn, Verse.AI.JobIssueParams jobParams) [0x00022] in <2464551dcee346df9bf15b1c06ef09e9>:0  
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug/Debug.bindings.h Line: 39)
```
The user specified they are on `RJW 5.5` and provide some more text: 

```
yes, also for comparison sake i tested it with only rjw and this mod (and harmony, hugslib) in a dev quicktest, spawned a bunch of pawns to speed it up, then waited, nothing. Im pretty sure its a mod conflict, i will update once i found out more (also i dont really know much about github and such, i had to google how to embed code like that in discord)
[[ ... ]]
so i tested it for longer and again it appeared with only a few rjw submods, hugslog if thats whats neede: 
https://gist.github.com/HugsLibRecordKeeper/60a3dd4446aafad7cb3fda15c88b8093
but it definitely happened alot less than my main save (only once sofar), so there probably some other mod that i use that speeds something up, but isnt the cause, anyway its not like it was logspamming or freezing my game so wasnt gamebreaking but odd nonetheless
```

[Hugslog](https://gist.github.com/HugsLibRecordKeeper/60a3dd4446aafad7cb3fda15c88b8093)

**Comment by vegapnk** [comment]
I got a hint from a user on Discord: 

[Suggested Place](https://github.com/vegapnk/Cumpilation/blob/55fa0e9c95ad22b7f7644b7d70e5d4eb36b6e8d0/1.5/Source/Gathering/Patches/Patch_CasualSexHelper_FindNearbyBucketToMasturbate.cs#L46) , which might just need a null-check.

**Comment by vegapnk** [comment]
Got another report: 

```
Am getting a repeating nullreference exception seemingly related to the cumbucket. I do have an advanced one placed down.
```

And [This log](https://pastebin.com/iZS5K0bN) with this error: 

```
Exception in rjw.ThinkNode_ChancePerHour_Fappin TryIssueJobPackage: System.NullReferenceException: Object reference not set to an instance of an object
[Ref 2E16C013]
 at rjw.ISexPartHediffExtensions.GetPartComp (rjw.ISexPartHediff partHediff) [0x00000] in <36fdf7cfc73c41228e347ef9cdc14d0d>:0 
 at Cumpilation.Gathering.Patch_CasualSexHelper_FindNearbyBucketToMasturbate.Prefix (Verse.Pawn pawn, Verse.Pawn partner, Verse.IntVec3& __result) [0x000c9] in <59f6c4056e6f4708a47ba91679f74dbb>:0 
 at rjw.CasualSex_Helper.FindSexLocation (Verse.Pawn pawn, Verse.Pawn partner) [0x00024] in <36fdf7cfc73c41228e347ef9cdc14d0d>:0 
     - PREFIX vegapnk.cumpilation: Boolean Cumpilation.Gathering.Patch_CasualSexHelper_FindNearbyBucketToMasturbate:Prefix(Pawn pawn, Pawn partner, IntVec3& __result)
 at rjw.JobGiver_Masturbate.TryGiveJob (Verse.Pawn pawn) [0x000f2] in <36fdf7cfc73c41228e347ef9cdc14d0d>:0 
 at Verse.AI.ThinkNode_JobGiver.TryIssueJobPackage (Verse.Pawn pawn, Verse.AI.JobIssueParams jobParams) [0x00008] in <f0ac5eb9b52e4cc396c70fc9a4ee15e5>:0 
     - POSTFIX Krkr.Rule56: Void CombatAI.Patches.ThinkNode_Patch+ThinkNode_TryIssueJobPackage_Patch:Postfix(ThinkNode __instance, ThinkResult __result, Pawn pawn)
 at (wrapper dynamic-method) MonoMod.Utils.DynamicMethodDefinition.Glue:AbiFixup<Verse.AI.ThinkResult
 [...]
 ```
 
 User says they are at commit 67a06a023656fb9ec5a02236f3f3ff8851de8402 
 But I believe I have not published a build since c6a1f80

**Comment by vegapnk** [comment]
Seems fixed.
